cli/internal/scope/scope.go

```diff
@@ -204,7 +204,8 @@ func getChangedPackages(changedFiles []string, packageInfos map[interface{}]*fs.
 	for _, changedFile := range changedFiles {
 		found := false
 		for pkgName, pkgInfo := range packageInfos {
-			if pkgName != util.RootPkgName && strings.HasPrefix(changedFile, pkgInfo.Dir) {
+			// Add trailing slash to avoid package substring conflicts: (e.g. react, react-dom)
+			if pkgName != util.RootPkgName && strings.HasPrefix(changedFile, pkgInfo.Dir+"/") {
```

Comment:
Hey Will! I noticed this breaks [one of the other tests](https://github.com/vercel/turborepo/blob/6e1f64ccceecfb98a42f926d48bedeb54b0b3b1c/cli/internal/scope/scope_test.go#L170) where the `changedFile` is `libs/libB` (no trailing slash)... TBH I'm not sure how realistic that case is, but it might be better to check each full path segment in the dirname.

I didn't want to step on your PR, I made a commit on a fork here: https://github.com/vercel/turborepo/commit/015cb34e9504669a36055a3b04f0f3ea29d1e9b3 

Feel free to cherry pick (if you can do that from a fork), or just pull in the code. It includes a few changes to the unit tests that I think are needed too (the unit tests take `packageInfos` as part of `ctx`)

Reply:
Ah nice catch, yeah was wondering if that case could come about. Would need at least a `package.json` for turbo to consider a directory a package, but then the directory wouldn't be the changed file, the `package.json` would. But maybe I'm not thinking of an edge case.

Using OS path separator is a good call, wasn't sure if these paths were OS-scoped. But looks like Nathan put up a PR so I'll close this.

Reply:
@wvanrensselaer You're 100% correct that it's weird, but (without checking) I suspect that's in there to protect from a weird symlink-to-directory edge case we always have to watch out for.